### PR TITLE
Updated Help Auto-Completion

### PR DESCRIPTION
**Pull request by benpturner**

### CODE REVIEW

File: poshc2/client/Help.py [hunk 1]
```diff
@@ -566,55 +556,54 @@
...
+#pre_help
+for line in pre_help.splitlines():
+	try:
+		entry = line.split(None, 1)[0]
+		if entry not in PRECOMMANDS and not any(char in special_characters for char in entry):
+			PRECOMMANDS.append(entry)
+	except IndexError:
+		pass
+	except:
+		print("Error")
```
Comment (riskydissonance) [inline poshc2/client/Help.py]: For each of these lets print the exception & traceback etc so we get info as to why its failing if it starts to do so.
Something like:

```
print_bad("Error building Sharp help")
traceback.print_exc()
```

File: poshc2/client/Help.py [hunk 2]
```diff
@@ -566,55 +556,54 @@
...
+special_characters = "!@#$%^&*()+=."
+
+#pre_help
+for line in pre_help.splitlines():
+	try:
+		entry = line.split(None, 1)[0]
+		if entry not in PRECOMMANDS and not any(char in special_characters for char in entry):
+			PRECOMMANDS.append(entry)
+	except IndexError:
+		pass
```
Comment (riskydissonance) [inline poshc2/client/Help.py]: As we're accessing the first index here the only time we'll get an IndexError is if the line is empty.
A better way of doing this then would be to check if the line is empty and if not add it, as opposed to trying to access something that may not exist and catching the error.

E.g. 

```
for line in pre_help.splitlines():
    try:
        line = line.strip()
        if line:
            entry = line.split(None, 1)[0]
            ...
    except Exception:
        print_bad("Error building prehelp")
        traceback.print_exc()
```

Without the except IndexError

**Comment by riskydissonance** [comment]
Loving this change

**Comment by benpturner** [comment]
Update as per comments and fixed conflict

**Review by riskydissonance — APPROVED** [review]

**Pull request merged**